### PR TITLE
conmon: fsync the log file

### PR DESCRIPTION
**Pull request by mrunalp**
@giuseppe @runcom @umohnani8 PTAL

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

File: conmon/conmon.c [hunk 1]
```diff
@@ -156,6 +156,13 @@ static int log_fd = -1;
 		exit(EXIT_FAILURE); \
 	} while (0)
 
+#define pwarn(s) \
+	do { \
+		fprintf(stderr, "[conmon:e]: %s %s\n", s, strerror(errno)); \
```
Comment (umohnani8) [inline conmon/conmon.c]: Might want to change this to `[conmon:w]` since it is a warning.

File: conmon/conmon.c [hunk 2]
```diff
@@ -156,6 +156,13 @@ static int log_fd = -1;
...
+		if (opt_syslog) \
+			syslog(LOG_ERR, "conmon %.20s <error>: %s %s\n", opt_cid, s, strerror(errno)); \
```
Comment (umohnani8) [inline conmon/conmon.c]: Same here, `LOG_WARN` and `<pwarn>` instead.

File: conmon/conmon.c (hunk 1 above)
Comment (umohnani8) [reply conmon/conmon.c]: Also do we want t check if the log level is set to warn and above, or do we just always want to log this?

Comment (mrunalp) [reply conmon/conmon.c]: We always want to log this.

File: conmon/conmon.c (hunk 2 above)
Comment (mrunalp) [reply conmon/conmon.c]: There is no `LOG_WARN` so using `LOG_INFO` instead. Thanks!

File: conmon/conmon.c [hunk 3]
```diff
@@ -398,6 +405,11 @@ static void reopen_log_file(void)
 {
 	_cleanup_free_ char *opt_log_path_tmp = g_strdup_printf("%s.tmp", opt_log_path);
 
+	/* Sync the logs to disk */
+	if (fsync(log_fd) < 0) {
```
Comment (giuseppe) [inline conmon/conmon.c]: we are going to create a new file few lines below, so this should not make a difference (unless the file cannot be re-opened), either way it doesn't hurt

**Comment by giuseppe** [comment]
LGTM

**Comment by umohnani8** [comment]
One question, but LGTM

**Pull request merged**